python/flatbuffers/builder.py

```diff
@@ -371,20 +371,21 @@ def StartVector(self, elemSize, numElems, alignment):
 
         self.assertNotNested()
         self.nested = True
+        self.vectorNumElems = numElems
         self.Prep(N.Uint32Flags.bytewidth, elemSize*numElems)
         self.Prep(alignment, elemSize*numElems)  # In case alignment > int.
         return self.Offset()
     ## @endcond
 
-    def EndVector(self, vectorNumElems):
+    def EndVector(self, vectorNumElems = None):
         """EndVector writes data necessary to finish vector construction."""
 
         self.assertNested()
         ## @cond FLATBUFFERS_INTERNAL
         self.nested = False
         ## @endcond
         # we already made space for this, so write without PrependUint32
-        self.PlaceUOffsetT(vectorNumElems)
```

Comment:
This can just always be `self.vectorNumElems`, and the function argument can be ignored, since its always set on the builder?

In fact, I'd almost be ok with just removing the argument, since we are lining up quite a few breaking changes for the next release.

Reply:
@rw

Reply:
The problem is in Builder EndVector is called a lot without calling StartVector (such as in CreateString and CreateByteVector).

We could instead just manually set self.vectorNumElems before each of those calls, but that makes things more complicated than just leaving an option to call EndVector with an argument.

Reply:
why is that more complicated? this is entirely in implementation code. important is that we don't have unnecessary conditionals in frequently executed code.. and I think this will overall be cleaner.

Reply:
It's harder to tell what's going on when a function like CreateString sets self.vectorNumElems to X and then calls EndVector vs. just calling EndVector(X). That said, you're right that this is all implementation code and the other way would be slower.

This should be fixed now.